### PR TITLE
Support GPU image for dnntrainer component

### PR DESCRIPTION
**Pull request by hongye-sun**
Changes:
* Parameterize dnntrainer build script to support build gpu image
* Update dnntrainer dockerfile to base from tensorflow image
* Update build script to push gpu image to gcr and update tags in release script

Future changes:
* Update tfx taxi predict sample and notebook sample to support gpu mode

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/530)

### CODE REVIEW

File: components/kubeflow/dnntrainer/build_image.sh [hunk 1]
```diff
@@ -42,12 +50,18 @@ if [ -z "${TAG_NAME}" ]; then
   TAG_NAME=$(date +v%Y%m%d)-$(git describe --tags --always --dirty)-$(git diff | shasum -a256 | cut -c -6)
 fi
 
+TF_VERSION=1.6.0
```
Comment (Ark-kun) [inline components/kubeflow/dnntrainer/build_image.sh]: TF images have version, CPU/GPU and python version. Maybe we should just add the base image tag command line parameter.

Comment (hongye-sun) [reply components/kubeflow/dnntrainer/build_image.sh]: done

File: components/third_party_licenses.csv [hunk 2]
```diff
@@ -169,3 +169,11 @@ tensorflow-tensorboard,https://raw.githubusercontent.com/tensorflow/tensorboard/
 tensorflow-data-validation,https://raw.githubusercontent.com/tensorflow/data-validation/master/LICENSE,Apache 2.0
 tensorflow-metadata,https://raw.githubusercontent.com/tensorflow/metadata/master/LICENSE,Apache 2.0
 defusedxml,https://raw.githubusercontent.com/tiran/defusedxml/master/LICENSE,PSF
+backports.functools-lru-cache,https://raw.githubusercontent.com/jaraco/backports.functools_lru_cache/master/LICENSE,MIT
+cycler,https://raw.githubusercontent.com/matplotlib/cycler/master/LICENSE,MIT
+h5py,https://raw.githubusercontent.com/h5py/h5py/07f3c362887563c1040b10d2cbd8cca49ceac1b9/licenses/license.txt
```
Comment (Ark-kun) [inline components/third_party_licenses.csv]: https://raw.githubusercontent.com/h5py/h5py/master/licenses/license.txt

**Comment by Ark-kun** [comment]
BTW, what errors do you get when you try using the GPU image on a non-GPU cluster?
Normally TF supports both CPU and GPU.

File: components/third_party_licenses.csv (hunk 2 above)
Comment (hongye-sun) [reply components/third_party_licenses.csv]: done

**Comment by hongye-sun** [comment]
I vaguely remember that It's related to CUDA driver not found error.

**Comment by hongye-sun** [comment]
@qimingj could you give me another lgtm? thanks.

**Pull request merged**